homeworks/week2/Vu-Nguyen.txt

```diff
@@ -0,0 +1,50 @@
+package Leet.DSAClass.RemoveDuplicateCharacters;
+
+// Problem: Remove duplicate characters in an array of characters. The orders of the characters in the input array must be preserved.
+//
+// Hints:
+//  1. It is ok to start use some build-in library, but the end goal is to not using build-in library.
+//  2. If you are allowed to use O(1) memory, what is the best runtime you can get?
+//  3. If you are restricted to O(N) memory, what is the best runtime you can get?
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+class Main {
+    public static void main(String[] args) {
+        char[] s = {'h', 'e', 'l', 'l', 'o', ' ', 'w', 'o', 'r', 'l', 'd'};
```

Comment:
It's a good practice to make sure you've considered and tested different cases. For example, your function doesn't handle `null` input well.

Let's add them in the following exercises